plutus-context-builder/src/Test/Plutus/ContextBuilder.hs

```diff
@@ -141,407 +155,487 @@ import Test.Plutus.ContextBuilder.Minting (
 
- @since 1.0
+ @since 2.0
 -}
 output ::
   forall (p :: Purpose).
-  Text ->
   SideUTXO ->
-  ContextBuilder p
-output name x = mempty {cbOutputs = Map.singleton name x}
+  ContextBuilder p 'Anonymous
+output x = NoNames $ mempty {cfOutputs = pure x}
 
-{- | Single 'ValidatorUTXO'-output context.
+{- | Anonymous context from a single 'ValidatorUTXO' output.
 
- @since 1.0
+ @since 2.0
 -}
 validatorOutput ::
   forall (d :: Type) (r :: Type).
   (FromData d, ToData d, Show d) =>
   Text ->
   ValidatorUTXO d ->
-  ContextBuilder ( 'ForSpending d r)
+  ContextBuilder ( 'ForSpending d r) 'Anonymous
 validatorOutput name x =
-  mempty {cbValidatorOutputs = ValidatorUTXOs $ Map.singleton name x}
+  NoNames $ mempty {cfValidatorOutputs = ValidatorUTXOs $ Map.singleton name x}
 
-{- | Context with one signature.
+{- | Anonymous context signed with one signature.
 
- @since 1.0
+ @since 2.0
 -}
 signedWith ::
   forall (p :: Purpose).
-  Text ->
   PubKeyHash ->
-  ContextBuilder p
-signedWith name x = mempty {cbSignatures = Map.singleton name x}
+  ContextBuilder p 'Anonymous
+signedWith x = NoNames $ mempty {cfSignatures = pure x}
 
-{- | Context with one additional datum.
+{- | Anonymous context with one additional datum.
 
- @since 1.0
+ @since 2.0
 -}
 datum ::
   forall (p :: Purpose).
-  Text ->
   BuiltinData ->
-  ContextBuilder p
-datum name x = mempty {cbDatums = Map.singleton name x}
+  ContextBuilder p 'Anonymous
+datum x = NoNames $ mempty {cfDatums = pure x}
 
 {- | Short for @'datum' '.' 'toBuiltinData'@.
 
- @since 1.0
+ @since 2.0
 -}
 addDatum ::
   forall (p :: Purpose) (a :: Type).
   (ToData a) =>
-  Text ->
   a ->
-  ContextBuilder p
-addDatum name = datum name . toBuiltinData
+  ContextBuilder p 'Anonymous
+addDatum = datum . toBuiltinData
 
-{- | Context with 'Minting' value using a minting policy other than the tested one.
+{- | Anonymous context with a 'Minting' value, using a minting policy other than
+ the one being tested.
 
  = Note
 
- Do not use this for 'Value' being minted by the tested minting policy.
+ Do not use this for 'Value's being minted by the minting policy you want to
+ test with the 'ScriptContext' this would produce. Asset classes with
+ 'CurrencySymbol's matching 'testCurrencySymbol' in the 'TransactionConfig'
+ provided when building the 'ScriptContext' will be excluded.
 
- Asset classes with 'CurrencySymbol' matching testCurrencySymbol in 'TransactionConfig'
- will be excluded from the resulting 'ScriptContext'.
-
- @since 1.0
+ @since 2.0
 -}
 minting ::
   forall (p :: Purpose).
-  Text ->
   Minting ->
-  ContextBuilder p
-minting name x = mempty {cbMinting = Map.singleton name x}
+  ContextBuilder p 'Anonymous
+minting x = NoNames $ mempty {cfMinting = pure x}
 
-{- | Context with an output at the given public key with the given 'Value'.
+{- | Name an arbitrary sub-context. The argument order is designed for infix
+ use:
 
- @since 1.0
+ > subContext :: ContextBuilder p 'Named
```

Comment:
`subContext` or `named`?

Reply:
That's just an example with a name and type signature - the actual code is below.